### PR TITLE
Generate make helping message dynamicaly.

### PR DESCRIPTION
**Pull request by 4383**
## Summarize
Generate make helping message dynamicaly by using
python code snippet inside Makefile.

All commented make targets will be added to the
help message. To be added to the helping message
comment need to start with '## '.

These specials comments are detected by the python code.
Python code generate the helping output from these results.

Notice that this commit introduce a dependency with python (compatible python 2 and 3).

Signed-off-by: Hervé Beraud <hberaud@redhat.com>

## Sample output
My changes generate the following outputs:
```shell
$ make help                                                                                                                          
Usage: make <target>                                                                                                                                                                                                 
 * 'binaries' - Build podman                                                                                                                                                                                         
 * 'changelog' - Generate changelog                                                                                                                                                                                  
 * 'clean' - Clean artifacts                                                                                                                                                                                         
 * 'docs' - Generate documentation                                                                                                                                                                                   
 * 'gofmt' - Verify the source code gofmt                                                                                                                                                                            
 * 'install' - Install binaries to system locations                                                                                                                                                                  
 * 'integration' - Execute integration tests                                                                                                                                                                         
 * 'libpodimage' - Build the libpod image                                                                                                                                                                            
 * 'lint' - Execute the source code linter                                                                                                                                                                           
 * 'local-cross' - Cross local compilation                                                                                                                                                                           
 * 'perftest' - Build perf tests                                                                                                                                                                                     
 * 'podman' - Build with podman                                                                                                                                                                                      
 * 'podman-remote' - Build with podman on remote environment                                                                                                                                                         
 * 'podman-remote-darwin' - Build with podman on remote OSX environment                                                                                                                                              
 * 'run-perftest' - Build and run perf tests                                                                                                                                                                         
 * 'shell' - Run the builted image and attach a shell                                                                                                                                                                
 * 'test' - Run tests on builted image                                                                                                                                                                               
 * 'testunit' - Run unittest on the builted image     
```

You can test them by using the following commands:
```shell
$ git clone https://github.com/4383/libpod -b improve-makefile
$ cd libpod
$ make help # generate the previous output
...
```

## Warning
- These changes introduce a dependency with python 2 or 3.
- I doesn't comment all targets due to a lack of knowledge about them. Feel free to ask to me to complet these comments with your knowledges about them.

Cheers!

### CODE REVIEW

**Comment by rhatdan** [comment]
bot, add author to whitelist
LGTM
@mheon @baude @umohnani8 @giuseppe @vrothberg @TomSweeneyRedHat PTAL

File: Makefile [hunk 1]
```diff
@@ -164,10 +172,10 @@ integration.fedora:
 integration.centos:
 	DIST=CentOS sh .papr_prepare.sh
 
-shell: libpodimage
+shell: libpodimage ## Run the builted image and attach a shell
```
Comment (TomSweeneyRedHat) [inline Makefile]: s/buillted/built/

File: Makefile [hunk 2]
```diff
@@ -164,10 +172,10 @@ integration.fedora:
...
 	${CONTAINER_RUNTIME} run -e STORAGE_OPTIONS="--storage-driver=vfs" -e CGROUP_MANAGER=cgroupfs -e TESTFLAGS -e OCI_RUNTIME -e TRAVIS -it --privileged --rm -v ${CURDIR}:/go/src/${PROJECT} ${LIBPOD_IMAGE} sh
 
-testunit: libpodimage
+testunit: libpodimage ## Run unittest on the builted image
```
Comment (TomSweeneyRedHat) [inline Makefile]: s/builted/built/

File: Makefile [hunk 3]
```diff
@@ -146,16 +154,16 @@ clean:
...
 
-libpodimage:
+libpodimage: ## Build the libpod image
 	${CONTAINER_RUNTIME} build -t ${LIBPOD_IMAGE} .
 
 dbuild: libpodimage
 	${CONTAINER_RUNTIME} run --name=${LIBPOD_INSTANCE} --privileged -v ${PWD}:/go/src/${PROJECT} --rm ${LIBPOD_IMAGE} make all
 
-test: libpodimage
+test: libpodimage ## Run tests on builted image
```
Comment (TomSweeneyRedHat) [inline Makefile]: s/builted/built/

**Comment by TomSweeneyRedHat** [comment]
@4383 thanks for the PR.  I like the look/feel of this, thanks for posting the PR.  I found a few nits that I commented on.  The python dependency I'm a little cautious on, but good to hear it works with python 2 or 3.  I think adding a python dependency is OK, but I'll let others weigh in on that.

**Comment by 4383** [comment]
@TomSweeneyRedHat nits fixed! :)

**Comment by TomSweeneyRedHat** [comment]
LGTM, assuming happy tests and the Python dependency being hip with others.

**Comment by rhatdan** [comment]
/lgtm
Thanks @4383

**Pull request merged**